OpenGov-OpenData/ckanext-og_datatablesview
# [OD-1496] Create og datatable view

## [OD-1496](https://opengovinc.atlassian.net/browse/OD-1496)

## Description

CHHS has reported that the "Data Tables" view displays "None" when a numeric value is left empty in the resource file. This could result in misleading display of the data. For example, with covid data, CHHS will leave the entry blank if the number is below a certain amount. The public would see None when in reality, the number is so small it shouldn't be reported.

This PR separates out the "Data Tables" view into it's own plugin. Changes made to the plugin include:
-  an empty cell being displayed instead of "None" with empty values of type numeric
- the downloads button was renamed to "Export" along with the display of it being made configurable in view edit page (option called “Hide Export" button)
- the current config for “Display export buttons” was renamed to “Display Copy and Print buttons“.

**Important Files Changed:**
- ckanext/og_datatablesview/controller.py: The logic to replace "None" with empty files is added in the form of a static method "remove_None_values_from_numeric_fields" being called in "og_ajax" function.
- ckanext/og_datatablesview/public/datatablesview.js: "Download" button was renamed to "Export"
- ckanext/og_datatablesview/templates/og_datatables/datatables_form.html: "Hide Export" was added and "Display exports utton" renamed to "Display Copy and Print buttons" in configuration form page.

Note: Other changes include naming conventions to reference this plugin.

## Test Procedure

1. Clone this directory to where your plugins are stored on your local and checkout this branch `dvasilov/PERF-821/2020-07-09/create-og_datatable_view_2`
2. In this directory, run `pip install -r requirements.txt` and then run `python setup.py develop`
3. In your ini file, replace `datatablesview` with `og_datatablesview`
4. Start CKAN and create a new dataset with a resource that contains a blank field. NOTE: Ensure that the column where the blank field will be contains only numbers.
5. Create "OG Data Tables" view for the newly created resource.
6. Ensure you see options "Hide Export" and. "Display Copy and Print buttons".
7. Create view and then check a blank cell is displayed where the empty cell is.
8. Play around with filtering data to ensure that functionality still works.
9. Click the "Export" button make sure you are able to download the data.
10. Edit view and toggle options mentioned int Step 6. to ensure they are working.


## Approval Criteria 

- Remove `None` values when the data type of the column is set to numeric. Data types of columns can be configured using the data dictionary tab when editing resources. I've attached a video showing an example of updating data types with the data dictionary 
- Rename Downloads button to Export and make the display of the button configurable in view edit page. The config can be called “Hide Export button“
- Rename current config for “Display export buttons” to “Display Copy and Print buttons“

## Submitter Checklist
